### ISSUE TITLE
Crash when panel title length changes.

### ISSUE THREAD

**Issue by stolk** [vurtun/nuklear#180]
When I change the length of the panel title in nk_begin(), from frame to frame, I get  a crash in:

```
#0  nk_draw_list_add_text (list=0x2728600, font=0x0, rect=..., text=0x27bcde0 "walktosawmill", len=13, 
    font_height=10, fg=...) at ../graphics/nuklear.h:6419
#1  0x0000000000430026 in nk_convert (ctx=0x2726710, cmds=<optimized out>, vertices=<optimized out>, 
    elements=<optimized out>, config=0x2729e18) at ../graphics/nuklear.h:6572
```
where font is zero when executing 'nk_draw_list_push_image(list, font->texture);'

If I change the title, but keep the title length the same, I experience no crash.

**Comment by vurtun** [comment]
Hey and thanks for your issue. At the moment I don't really understand what you are trying to do. At this point in time there is not direct way to change the name of the window, since I use the name to identify the window. But you could just pass a different name to `nk_begin` which will create another window.

If you want to keep the window and _really_ want to change the title I would do `nk_find_window` and update the window name.

```c
struct nk_context *ctx;
struct nk_window *win = nk_find_window(ctx,  "window_title");
win->name = nk_murmur_hash(ctx, "new_title", nk_strlen("new_title"), NK_WINDOW_TITLE);
```
 I am not 100% sure there are no side effects at the moment so it is probably best to test it.

**Comment by Tirus42** [comment]
I have the same issue here.
I use nuklear to create a gui for a image viewing program.
There i use the window title to show the name and some extra information about the current image.
As the image changes, I create a new string with the information about the now shown image.
As nuklear is stateless, I expected that there will be no problem.

But some times (couln't reproduce it reliable) it crashes with the following failed assert while switching though the images:

fb: [...]/nuklear/nuklear.h:4387: void* 
nk_buffer_alloc(nk_buffer*, nk_buffer_allocation_type, nk_size, nk_size): Assertion `b->type == 
NK_BUFFER_DYNAMIC' failed.
Aborted (core dumped)

**Comment by itscool** [comment]
Not stateless. Just hides the state from you to allow "immediate mode" style.

**Comment by vurtun** [comment]
@Tirus42 Thanks when I have some time I will write a small demo and will try to replicate the bug. The assert tells me that a fixed buffer is full. But I don't really know the reason at the moment. Quick question do you '\0' terminate the title of the window? Since if you don't it just copies a huge string into the command buffer which could cause the assert, that is the only idea I have at the moment.

**Comment by Tirus42** [comment]
I use snprintf to create the string. And it appends a terminating \0.

**Comment by vurtun** [comment]
I modified the demo to change the title of the window as a test case. While this approach is absolutely not a good idea I did not encounter the problem so far. Basically by changing the name of the window you delete and create the window every frame (it gets linked and unlinked from a free list so not that bad memory wise). The bigger problem of this approach is that you cannot have Scrollbars, Tabs, Groups or any widgets with state (properties, edit boxes, ...) since each of them get cleared every frame. I would suggest using the code I proposed above to make sure the state is kept over all frames. As for of the buffer assert macro it could be that there is a bug inside the window freeing code but I tested it and so far it looks correct. I will look further into it as soon as I have more time.
